pkg/daemon/daemon.go

```diff
@@ -93,398 +55,304 @@ func New(
-	go wait.Until(dn.runWorker, time.Second, stopCh)
-	log.Log.Info("Started workers")
+	ns := &sriovnetworkv1.SriovNetworkNodeState{}
+	// init openstack info
+	if vars.PlatformType == consts.VirtualOpenStack {
+		ns, err = dn.HostHelpers.GetCheckPointNodeState()
+		if err != nil {
+			return err
+		}
 
-	for {
-		select {
-		case <-stopCh:
-			log.Log.V(0).Info("Run(): stop daemon")
-			return nil
-		case err, more := <-exitCh:
-			log.Log.Error(err, "got an error")
-			if more {
-				dn.refreshCh <- Message{
-					syncStatus:    consts.SyncStatusFailed,
-					lastSyncError: err.Error(),
-				}
+		if ns == nil {
+			err = dn.platformHelpers.CreateOpenstackDevicesInfo()
+			if err != nil {
+				return err
 			}
-			return err
+		} else {
+			dn.platformHelpers.CreateOpenstackDevicesInfoFromNodeStatus(ns)
 		}
 	}
-}
 
-func (dn *Daemon) runWorker() {
-	for dn.processNextWorkItem() {
+	// get interfaces
+	err = dn.getHostNetworkStatus(ns)
+	if err != nil {
+		funcLog.Error(err, "failed to get host network status on init")
+		return err
 	}
-}
 
-func (dn *Daemon) enqueueNodeState(obj interface{}) {
-	var ns *sriovnetworkv1.SriovNetworkNodeState
-	var ok bool
-	if ns, ok = obj.(*sriovnetworkv1.SriovNetworkNodeState); !ok {
-		utilruntime.HandleError(fmt.Errorf("expected SriovNetworkNodeState but got %#v", obj))
-		return
+	// init vendor plugins
+	dn.loadedPlugins, err = loadPlugins(ns, dn.HostHelpers, dn.disabledPlugins)
+	if err != nil {
+		funcLog.Error(err, "failed to enable vendor plugins")
+		return err
 	}
-	key := ns.GetGeneration()
-	dn.workqueue.Add(key)
-}
-
-func (dn *Daemon) processNextWorkItem() bool {
-	log.Log.V(2).Info("processNextWorkItem", "worker-queue-size", dn.workqueue.Len())
-	obj, shutdown := dn.workqueue.Get()
-	if shutdown {
-		return false
-	}
-
-	log.Log.V(2).Info("get item from queue", "item", obj.(int64))
-
-	// We wrap this block in a func so we can defer c.workqueue.Done.
-	err := func(obj interface{}) error {
-		// We call Done here so the workqueue knows we have finished
-		// processing this item.
-		defer dn.workqueue.Done(obj)
-		var key int64
-		var ok bool
-		if key, ok = obj.(int64); !ok {
-			// As the item in the workqueue is actually invalid, we call
-			// Forget here.
-			dn.workqueue.Forget(obj)
-			utilruntime.HandleError(fmt.Errorf("expected workItem in workqueue but got %#v", obj))
-			return nil
-		}
-
-		err := dn.nodeStateSyncHandler()
-		if err != nil {
-			// Ereport error message, and put the item back to work queue for retry.
-			dn.refreshCh <- Message{
-				syncStatus:    consts.SyncStatusFailed,
-				lastSyncError: err.Error(),
-			}
-			<-dn.syncCh
-			dn.workqueue.AddRateLimited(key)
-			return fmt.Errorf("error syncing: %s, requeuing", err.Error())
-		}
-		// Finally, if no error occurs we Forget this item so it does not
-		// get queued again until another change happens.
-		dn.workqueue.Forget(obj)
-		log.Log.Info("Successfully synced")
-		return nil
-	}(obj)
 
+	// save init state
```

Comment:
nit: // save init state if it does not exist on host

Reply:
from the code I can see that we always write the file. I am missing something?